langchain/text_splitter.py

```diff
@@ -439,3 +440,261 @@ def __init__(self, **kwargs: Any):
             "",
         ]
         super().__init__(separators=separators, **kwargs)
+
+
+class Language(str, Enum):
```

Comment:
This looks good to me. 

Have you considered using a union of literals instead of an enum?

```
Language = Union[Literal['go'], Literal['js'], ...]
```

Gives the same static typing guarantees and makes it such that a user does not have to import an extra enum.

I think the only downside is that it's harder to list all the options in the set.

Reply:
Hmm I would think prompting available languages is more important to users

Reply:
I believe VSCode/PyLance would still prompt appropriately for a union of literals, but the enum is more foolproof.